contracts/apps/accounting/AccountingApp.sol

```diff
@@ -0,0 +1,186 @@
+        // For now the current period is the most recent one
+        return accountingPeriods.length - 1;
+    }
+
+    function getCurrentAccountingPeriod() returns (address baseToken, bytes2 ct_hour, bytes2 ct_day, bytes2 ct_month, bytes2 ct_weekday){
+        AccountingPeriod memory ap = accountingPeriods[getCurrentAccountingPeriodId()];
+        return (ap.baseToken, ap.ct_hour, ap.ct_day, ap.ct_month, ap.ct_weekday);
+    }
+
+    function startNextAccountingPeriod() {
+        AccountingPeriod memory ap = defaultAccountingPeriodSettings;
+        accountingPeriods.push(ap);
+    }
+
+    function setDefaultAccountingPeriodSettings(address baseToken, bytes2 ct_hour, bytes2 ct_day, bytes2 ct_month, bytes2 ct_weekday) {
+        defaultAccountingPeriodSettings.baseToken = baseToken;
+        defaultAccountingPeriodSettings.ct_hour = ct_hour;
+        defaultAccountingPeriodSettings.ct_day = ct_day;
+        defaultAccountingPeriodSettings.ct_month = ct_month;
+        defaultAccountingPeriodSettings.ct_weekday = ct_weekday;
+    }
+
+
+    //
+    // TRANSACTIONS
+    //
+
+    // The concept of sending tokens to or from the org
+    struct Transaction {
+        address token;
+        int value;
+        address baseToken;
+        int baseValue;
+        address externalAddress;
+        string reference;
+        uint timestamp;
+        uint accountingPeriodId;  // in which accounting period did this occur
+    }    
+
+    // The state a transaction update can be.
+    // New states should be added to the end to maintain the 
+    // order of the index when interfacing with web3.
+    enum TransactionState {
+        New, // not needed?
+        PendingApproval,
+        Failed,
+        Succeeded
+    }
+
+    // The change in Transaciton state over time
+    struct TransactionUpdate {
+        uint transactionId; // Parent Transaction
+        TransactionState state;
+        string reason;
+        address actor; // who performed this update 
+    }    
+
+    Transaction[] public transactions;
+    TransactionUpdate[] public transactionUpdates;
+
+    // Reverse relation of a Transaction ID  -> TransactionsUpdatesIds[]
+    // transactionUpdatesRelation[tid] = [tuid_0..tuid_N]
+    mapping (uint => uint[]) public transactionUpdatesRelation;
+
+    // Create a new transaction and return the id of the new transaction.
+    // externalAddress is where the transication is coming or going to.
+    bytes4 constant NEW_TRANSACTION_SID = bytes4(sha3('newTransaction(address,int,address,string,TransactionState)'));
+    function newTransaction(address token, int value, address externalAddress, string reference, TransactionState initialState) returns (uint) {
+
+        uint tid = transactions.push(Transaction({
+            externalAddress: externalAddress, 
+            token: token, 
+            value: value, 
+            // TODO: get base token and exchange rate from oracle 
```

Comment:
base token should be gotten from the current accounting period base token, right?